### PR TITLE
Basic svg rendering structure

### PR DESCRIPTION
**Pull request by ToBinio**
This PR adds the basic file and code structure for the `SVGRendering` struct

note:
`forms/entity_type` is only a placeholder until the real types are created

### CODE REVIEW

File: shellborn/src/render/svg/svg_renderer.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    pub fn add_entity(&mut self, entity: &Positioned<Entity>) {
+        let width = 100;
+        let height = 50;
+
+        self.update_view_box(&ViewBox::new(
+            entity.center_x as i32 - width / 2,
+            entity.center_y as i32 - height / 2,
+            entity.center_x as i32 + width / 2,
+            entity.center_y as i32 + height / 2,
+        ));
```
Comment (JonasPfeifer05) [inline shellborn/src/render/svg/svg_renderer.rs:33]: maybe change the function to accpet anything that implements the trait AsRef<ViewBox>. Would remove the & but isnt necessary

Comment (JonasPfeifer05) [reply shellborn/src/render/svg/svg_renderer.rs:33]: or implement the trait Into<Viewbox> for the Positioned struct

File: shellborn/src/render/svg/render_able.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+use crate::entity::Entity;
+use crate::positioned::Positioned;
+use crate::render::svg::view_box::ViewBox;
+use svg::node::element::Ellipse;
+use svg::Document;
+
+pub trait SvgRenderAble {
+    fn render(&self, document: Document) -> Document;
```
Comment (JonasPfeifer05) [inline shellborn/src/render/svg/render_able.rs:8]:
```suggestion
    fn render_onto(&self, document: Document) -> Document;
```

**Review by JonasPfeifer05 — CHANGES_REQUESTED** [review]
everything else is ok

File: shellborn/src/render/svg/svg_renderer.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    fn default() -> Self {
+        SVGRenderer {
+            view_box: ViewBox::default(),
+            document: Document::new(),
+        }
+    }
+}
+
+impl SVGRenderer {
+    pub fn add(&mut self, render_able: &dyn SvgRenderAble) {
```
Comment (JonasPfeifer05) [inline shellborn/src/render/svg/svg_renderer.rs:23]:
```suggestion
    pub fn append(&mut self, render_able: &dyn SvgRenderAble) {
```

**Review by JonasPfeifer05 — CHANGES_REQUESTED** [review]

File: shellborn/src/render/svg/render_able.rs (hunk 2 above)
Comment (ToBinio) [reply shellborn/src/render/svg/render_able.rs]: done

File: shellborn/src/render/svg/svg_renderer.rs (hunk 3 above)
Comment (ToBinio) [reply shellborn/src/render/svg/svg_renderer.rs]: done

**Review by JonasPfeifer05 — APPROVED** [review]
ok 🐚

**Pull request merged**